File: backend/src/main/java/t/lab/guide/service/mock/MockPointService.java
package t.lab.guide.service.mock;

import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Service;
import org.springframework.web.multipart.MultipartFile;
import t.lab.guide.dto.admin.point.AdminCreatePointRequest;
import t.lab.guide.dto.admin.point.AdminPatchPointRequest;
import t.lab.guide.dto.admin.point.AdminPointDetailResponse;
import t.lab.guide.dto.admin.point.AdminPointMediaItem;
import t.lab.guide.dto.admin.point.AdminPointPageResponse;
import t.lab.guide.dto.admin.point.AdminPointShortItem;
import t.lab.guide.dto.admin.point.AdminUploadPointMediaRequest;
import t.lab.guide.dto.common.GeoPoint;
import t.lab.guide.dto.point.PointDetailResponse;
import t.lab.guide.dto.point.PointListResponse;
import t.lab.guide.dto.point.PointMediaItem;
import t.lab.guide.dto.point.PointSearchRequest;
import t.lab.guide.dto.point.PointShortItem;
import t.lab.guide.dto.point.category.CategoryItem;
import t.lab.guide.entity.enums.MediaType;
import t.lab.guide.exception.NotFoundException;
import t.lab.guide.service.PointService;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.List;
import java.util.stream.IntStream;

@Service
@Profile("demo")
public class MockPointService implements PointService {

    private static final List<CategoryItem> categories = List.of(
            new CategoryItem(1L, "Достопримечательности", "attraction"),
            new CategoryItem(2L, "Шопинг", "shop"),
            new CategoryItem(3L, "Музеи", "museum"),
            new CategoryItem(4L, "Парки", "park"),
            new CategoryItem(5L, "Еда", "food")
    );

    private static final List<PointShortItem> mock = List.of(
            new PointShortItem(1L, "Красная площадь", 1L, "Достопримечательности",
                    new GeoPoint(new BigDecimal("55.753544"), new BigDecimal("37.621202")), 60),
            new PointShortItem(2L, "ГУМ", 2L, "Шопинг",
                    new GeoPoint(new BigDecimal("55.754713"), new BigDecimal("37.621500")), 90),
            new PointShortItem(3L, "Третьяковская галерея", 3L, "Музеи",
                    new GeoPoint(new BigDecimal("55.741426"), new BigDecimal("37.620137")), 120),
            new PointShortItem(4L, "Парк Горького", 4L, "Парки",
                    new GeoPoint(new BigDecimal("55.729812"), new BigDecimal("37.601348")), 150),
            new PointShortItem(5L, "Кафе Пушкинъ", 5L, "Еда",
                    new GeoPoint(new BigDecimal("55.764804"), new BigDecimal("37.604392")), 90)
    );

    private static final List<PointDetailResponse> mockDetail = List.of(
            new PointDetailResponse(
                    1L,
                    "Красная площадь",
                    "Главная площадь Москвы, расположенная в самом центре города между Московским Кремлём и Китай-городом. Включена в список Всемирного наследия ЮНЕСКО.",
                    1L,
                    "Достопримечательности",
                    "Москва, Красная площадь",
                    new GeoPoint(new BigDecimal("55.753544"), new BigDecimal("37.621202")),
                    60,
                    "Круглосуточно",
                    List.of(
                            new PointMediaItem("https://cdn.t-guide.mock/points/1/photo-1.jpg", MediaType.PHOTO, 0),
                            new PointMediaItem("https://cdn.t-guide.mock/points/1/photo-2.jpg", MediaType.PHOTO, 1),
                            new PointMediaItem("https://cdn.t-guide.mock/points/1/audio-guide.mp3", MediaType.AUDIO, 2)
                    )
            ),
            new PointDetailResponse(
                    2L,
                    "ГУМ",
                    "Главный универсальный магазин — крупнейший торговый комплекс Москвы, расположенный на Красной площади. Памятник архитектуры псевдорусского стиля.",
                    2L,
                    "Шопинг",
                    "Москва, Красная площадь, 3",
                    new GeoPoint(new BigDecimal("55.754713"), new BigDecimal("37.621500")),
                    90,
                    "10:00 - 22:00",
                    List.of(
                            new PointMediaItem("https://cdn.t-guide.mock/points/2/photo-1.jpg", MediaType.PHOTO, 0),
                            new PointMediaItem("https://cdn.t-guide.mock/points/2/photo-2.jpg", MediaType.PHOTO, 1)
                    )
            ),
            new PointDetailResponse(
                    3L,
                    "Третьяковская галерея",
                    "Художественный музей в Москве, основанный в 1856 году купцом Павлом Третьяковым. Имеет одну из самых крупных в мире коллекций русского изобразительного искусства.",
                    3L,
                    "Музеи",
                    "Москва, Лаврушинский переулок, 10",
                    new GeoPoint(new BigDecimal("55.741426"), new BigDecimal("37.620137")),
                    120,
                    "10:00 - 18:00, выходной — понедельник",
                    List.of(
                            new PointMediaItem("https://cdn.t-guide.mock/points/3/photo-1.jpg", MediaType.PHOTO, 0),
                            new PointMediaItem("https://cdn.t-guide.mock/points/3/video-tour.mp4", MediaType.VIDEO, 1),
                            new PointMediaItem("https://cdn.t-guide.mock/points/3/audio-guide.mp3", MediaType.AUDIO, 2)
                    )
            ),
            new PointDetailResponse(
                    4L,
                    "Парк Горького",
                    "Центральный парк культуры и отдыха в Москве. Современное городское пространство с набережной, велодорожками, кафе и лекториями.",
                    4L,
                    "Парки",
                    "Москва, ул. Крымский Вал, 9",
                    new GeoPoint(new BigDecimal("55.729812"), new BigDecimal("37.601348")),
                    150,
                    "Круглосуточно",
                    List.of(
                            new PointMediaItem("https://cdn.t-guide.mock/points/4/photo-1.jpg", MediaType.PHOTO, 0),
                            new PointMediaItem("https://cdn.t-guide.mock/points/4/photo-2.jpg", MediaType.PHOTO, 1),
                            new PointMediaItem("https://cdn.t-guide.mock/points/4/video-aerial.mp4", MediaType.VIDEO, 2)
                    )
            ),
            new PointDetailResponse(
                    5L,
                    "Кафе Пушкинъ",
                    "Знаменитый московский ресторан русской кухни в стиле дворянской усадьбы XIX века. Расположен на Тверском бульваре.",
                    5L,
                    "Еда",
                    "Москва, Тверской бульвар, 26А",
                    new GeoPoint(new BigDecimal("55.764804"), new BigDecimal("37.604392")),
                    90,
                    "Круглосуточно",
                    List.of(
                            new PointMediaItem("https://cdn.t-guide.mock/points/5/photo-1.jpg", MediaType.PHOTO, 0),
                            new PointMediaItem("https://cdn.t-guide.mock/points/5/photo-interior.jpg", MediaType.PHOTO, 1)
                    )
            )
    );

    @Override
    public PointListResponse searchPoints(PointSearchRequest request) {
        var categorySlugs = request.categorySlugs();
        var visitTime = request.visitTime();

        var filterIds = categorySlugs == null ? List.<Long>of() : categories.stream()
                .filter(c -> categorySlugs.contains(c.slug()))
                .map(CategoryItem::id)
                .toList();

        var filtered = mock.stream()
                .filter(p -> matchesCategories(p.categoryId(), categorySlugs, filterIds))
                .filter(p -> matchesDuration(p.visitTime(), visitTime))
                .toList();

        return new PointListResponse(filtered);
    }

    private static boolean matchesCategories(Long pointCategoryId, List<String> filterSlugs, List<Long> filterIds) {
        if (filterSlugs == null || filterSlugs.isEmpty()) {
            return true;
        }
        return pointCategoryId != null && filterIds.contains(pointCategoryId);
    }

    private static boolean matchesDuration(Integer pointVisitTime, Integer maxVisitTime) {
        if (maxVisitTime == null) {
            return true;
        }
        return pointVisitTime != null && pointVisitTime <= maxVisitTime;
    }

    @Override
    public PointDetailResponse getPointDetail(Long id) {
        return mockDetail.stream()
                .filter(p -> p.id().equals(id))
                .findFirst()
                .orElseThrow(() -> new NotFoundException("Точка с id=" + id + " не найдена"));
    }

    @Override
    public AdminPointPageResponse getPointsPage(int page, int size, String sortBy, String sortDirection, String search){
        long totalElements = 247;
        page = Math.max(page, 0);
        size = Math.min(Math.max(size, 0), 100);
        int totalPages = size == 0 ? 0 : (int) Math.ceil((double) totalElements / size);
        int from = page * size;
        int to = (int) Math.min(from + size, totalElements);
        int count = Math.max(0, to - from);

        List<AdminPointShortItem> points = IntStream.range(0, count)
                .mapToObj(i -> {
                    long id = from + i + 1L;
                    CategoryItem category = categories.get((int) ((id - 1) % categories.size()));
                    Integer visitTime = 30 + (int) ((id % 6) * 15);
                    boolean isActive = id % 11 != 0;
                    return new AdminPointShortItem(
                            id,
                            "Точка интереса №" + id,
                            category.id(),
                            category.name(),
                            visitTime,
                            isActive,
                            OffsetDateTime.now().minusDays(id)
                    );
                })
                .toList();

        return new AdminPointPageResponse(points, page, size, totalElements, totalPages);
    }

    @Override
    public AdminPointDetailResponse getAdminPointDetail(Long id) {
        PointDetailResponse detail = getPointDetail(id);
        List<AdminPointMediaItem> adminMedia = IntStream.range(0, detail.media().size())
                .mapToObj(i -> {
                    PointMediaItem m = detail.media().get(i);
                    return new AdminPointMediaItem(
                            (long) (i + 1),
                            detail.id(),
                            m.url(),
                            m.type(),
                            m.sortOrder(),
                            OffsetDateTime.now().minusDays(i + 1)
                    );
                })
                .toList();
        return AdminPointDetailResponse.builder()
                .id(detail.id())
                .title(detail.title())
                .description(detail.description())
                .categoryId(detail.categoryId())
                .categoryName(detail.categoryName())
                .address(detail.address())
                .coordinates(detail.coordinates())
                .visitTime(detail.visitTime())
                .workingHours(detail.workingHours())
                .isActive(true)
                .media(adminMedia)
                .createdAt(OffsetDateTime.now().minusDays(30))
                .updatedAt(OffsetDateTime.now().minusDays(1))
                .build();
    }

    @Override
    public AdminPointDetailResponse createPoint(AdminCreatePointRequest request) {
        long newId = mock.size() + 1L;
        String categoryName = categories.stream()
                .filter(c -> c.id().equals(request.categoryId()))
                .map(CategoryItem::name)
                .findFirst()
                .orElseThrow(() -> new NotFoundException(
                        "Категория с id=" + request.categoryId() + " не найдена"));
        OffsetDateTime now = OffsetDateTime.now();
        return AdminPointDetailResponse.builder()
                .id(newId)
                .title(request.title())
                .description(request.description())
                .categoryId(request.categoryId())
                .categoryName(categoryName)
                .address(request.address())
                .coordinates(request.coordinates())
                .visitTime(request.visitTime())
                .workingHours(request.workingHours())
                .isActive(request.isActive() == null || request.isActive())
                .media(List.of())
                .createdAt(now)
                .updatedAt(now)
                .build();
    }

    @Override
    public AdminPointDetailResponse patchPoint(Long id, AdminPatchPointRequest request) {
        AdminPointDetailResponse current = getAdminPointDetail(id);
        Long categoryId = request.categoryId() != null ? request.categoryId() : current.categoryId();
        String categoryName = request.categoryId() != null
                ? categories.stream()
                    .filter(c -> c.id().equals(request.categoryId()))
                    .map(CategoryItem::name)
                    .findFirst()
                    .orElseThrow(() -> new NotFoundException(
                            "Категория с id=" + request.categoryId() + " не найдена"))
                : current.categoryName();
        return AdminPointDetailResponse.builder()
                .id(current.id())
                .title(request.title() != null ? request.title() : current.title())
                .description(request.description() != null ? request.description() : current.description())
                .categoryId(categoryId)
                .categoryName(categoryName)
                .address(request.address() != null ? request.address() : current.address())
                .coordinates(request.coordinates() != null ? request.coordinates() : current.coordinates())
                .visitTime(request.visitTime() != null ? request.visitTime() : current.visitTime())
                .workingHours(request.workingHours() != null ? request.workingHours() : current.workingHours())
                .isActive(request.isActive() != null ? request.isActive() : current.isActive())
                .media(current.media())
                .createdAt(current.createdAt())
                .updatedAt(OffsetDateTime.now())
                .build();
    }

    @Override
    public void deletePoint(Long id) {
        mockDetail.stream()
                .filter(p -> p.id().equals(id))
                .findFirst()
                .orElseThrow(() -> new NotFoundException("Точка с id=" + id + " не найдена"));
    }

    @Override
    public AdminPointMediaItem uploadPointMedia(Long pointId, MultipartFile file, AdminUploadPointMediaRequest request) {
        getPointDetail(pointId);
        String fileName = file != null && file.getOriginalFilename() != null
                ? file.getOriginalFilename()
                : "upload.bin";
        String mockUrl = "https://cdn.t-guide.mock/points/" + pointId + "/" + fileName;
        return new AdminPointMediaItem(
                System.currentTimeMillis(),
                pointId,
                mockUrl,
                request.type(),
                request.sortOrder() != null ? request.sortOrder() : 0,
                OffsetDateTime.now()
        );
    }

    @Override
    public void deletePointMedia(Long pointId, Long mediaId) {
        getPointDetail(pointId);
    }

}

File: backend/src/main/kotlin/t/lab/guide/service/mock/MockPointService.kt
package t.lab.guide.service.mock

import org.springframework.context.annotation.Profile
import org.springframework.stereotype.Service
import org.springframework.web.multipart.MultipartFile
import t.lab.guide.dto.admin.point.AdminPatchPointRequest
import t.lab.guide.dto.admin.point.AdminPointDetailResponse
import t.lab.guide.dto.admin.point.AdminPointMediaItem
import t.lab.guide.dto.admin.point.AdminPointPageResponse
import t.lab.guide.dto.admin.point.AdminPointShortItem
import t.lab.guide.dto.admin.point.command.AdminCreatePointCommand
import t.lab.guide.dto.admin.point.command.AdminUploadPointMediaCommand
import t.lab.guide.dto.category.CategoryItem
import t.lab.guide.dto.common.GeoPoint
import t.lab.guide.dto.common.command.toDto
import t.lab.guide.dto.point.PointDetailResponse
import t.lab.guide.dto.point.PointListResponse
import t.lab.guide.dto.point.PointMediaItem
import t.lab.guide.dto.point.PointShortItem
import t.lab.guide.dto.point.command.PointSearchCommand
import t.lab.guide.enums.AdminPointSortField
import t.lab.guide.enums.MediaType
import t.lab.guide.enums.SortDirection
import t.lab.guide.exception.NotFoundException
import t.lab.guide.service.PointService
import java.math.BigDecimal
import java.time.OffsetDateTime

@Service
@Profile("demo")
class MockPointService : PointService {
    override fun searchPoints(request: PointSearchCommand): PointListResponse {
        val categorySlugs = request.categorySlugs
        val maxVisitTime = request.visitTime
        val filterIds =
            if (categorySlugs.isEmpty()) {
                emptyList()
            } else {
                categories.filter { it.slug in categorySlugs }.map { it.id }
            }
        val filtered =
            mock
                .filter { categorySlugs.isEmpty() || it.categoryId in filterIds }
                .filter { maxVisitTime == null || (it.visitTime ?: 0) <= maxVisitTime }
        return PointListResponse(filtered)
    }

    override fun getPointDetail(id: Long): PointDetailResponse =
        mockDetail.firstOrNull { it.id == id }
            ?: throw NotFoundException("Точка с id=$id не найдена")

    override fun getPointsPage(
        page: Int,
        size: Int,
        sortBy: AdminPointSortField?,
        sortDirection: SortDirection?,
        search: String?,
    ): AdminPointPageResponse {
        val totalElements = 247L
        val safePage = page.coerceAtLeast(0)
        val safeSize = size.coerceIn(0, 100)
        val totalPages =
            if (safeSize == 0) {
                0
            } else {
                ((totalElements + safeSize - 1) / safeSize).toInt()
            }
        val from = safePage * safeSize
        val to = minOf((from + safeSize).toLong(), totalElements).toInt()
        val count = (to - from).coerceAtLeast(0)

        val points =
            (0 until count).map { i ->
                val id = (from + i + 1).toLong()
                val category = categories[((id - 1) % categories.size).toInt()]
                val visitTime = 30 + (id % 6).toInt() * 15
                val isActive = id % 11 != 0L
                AdminPointShortItem(
                    id = id,
                    title = "Точка интереса №$id",
                    categoryId = category.id,
                    categoryName = category.name,
                    visitTime = visitTime,
                    isActive = isActive,
                    createdAt = OffsetDateTime.now().minusDays(id),
                )
            }
        return AdminPointPageResponse(
            points = points,
            page = safePage,
            size = safeSize,
            totalElements = totalElements,
            totalPages = totalPages,
        )
    }

    override fun getAdminPointDetail(id: Long): AdminPointDetailResponse {
        val detail = getPointDetail(id)
        val adminMedia =
            detail.media.mapIndexed { i, m ->
                AdminPointMediaItem(
                    id = (i + 1).toLong(),
                    pointId = detail.id,
                    url = m.url,
                    type = m.type,
                    sortOrder = m.sortOrder,
                    createdAt = OffsetDateTime.now().minusDays((i + 1).toLong()),
                )
            }
        return AdminPointDetailResponse(
            id = detail.id,
            title = detail.title,
            description = detail.description,
            shortDescription = detail.shortDescription,
            categoryId = detail.categoryId,
            categoryName = detail.categoryName,
            address = detail.address,
            coordinates = detail.coordinates,
            visitTime = detail.visitTime,
            workingHours = detail.workingHours,
            isActive = true,
            media = adminMedia,
            createdAt = OffsetDateTime.now().minusDays(30),
            updatedAt = OffsetDateTime.now().minusDays(1),
        )
    }

    override fun createPoint(request: AdminCreatePointCommand): AdminPointDetailResponse {
        val newId = mock.size + 1L
        val categoryName =
            categories.firstOrNull { it.id == request.categoryId }?.name
                ?: throw NotFoundException("Категория с id=${request.categoryId} не найдена")
        val now = OffsetDateTime.now()
        return AdminPointDetailResponse(
            id = newId,
            title = request.title,
            description = request.description,
            shortDescription = request.shortDescription,
            categoryId = request.categoryId,
            categoryName = categoryName,
            address = request.address,
            coordinates = request.coordinates.toDto(),
            visitTime = request.visitTime,
            workingHours = request.workingHours,
            isActive = request.isActive,
            media = emptyList(),
            createdAt = now,
            updatedAt = now,
        )
    }

    override fun patchPoint(
        id: Long,
        request: AdminPatchPointRequest,
    ): AdminPointDetailResponse {
        val current = getAdminPointDetail(id)
        val newCategoryId = request.categoryId ?: current.categoryId
        val newCategoryName =
            if (request.categoryId != null) {
                categories.firstOrNull { it.id == request.categoryId }?.name
                    ?: throw NotFoundException("Категория с id=${request.categoryId} не найдена")
            } else {
                current.categoryName
            }
        return AdminPointDetailResponse(
            id = current.id,
            title = request.title ?: current.title,
            description = request.description ?: current.description,
            shortDescription = request.shortDescription ?: current.shortDescription,
            categoryId = newCategoryId,
            categoryName = newCategoryName,
            address = request.address ?: current.address,
            coordinates = request.coordinates ?: current.coordinates,
            visitTime = request.visitTime ?: current.visitTime,
            workingHours = request.workingHours ?: current.workingHours,
            isActive = request.isActive ?: current.isActive,
            media = current.media,
            createdAt = current.createdAt,
            updatedAt = OffsetDateTime.now(),
        )
    }

    override fun deletePoint(id: Long) {
        mockDetail.firstOrNull { it.id == id }
            ?: throw NotFoundException("Точка с id=$id не найдена")
    }

    override fun uploadPointMedia(
        pointId: Long,
        file: MultipartFile,
        request: AdminUploadPointMediaCommand,
    ): AdminPointMediaItem {
        getPointDetail(pointId)
        val fileName = file.originalFilename ?: "upload.bin"
        val mockUrl = "https://cdn.t-guide.mock/points/$pointId/$fileName"
        return AdminPointMediaItem(
            id = System.currentTimeMillis(),
            pointId = pointId,
            url = mockUrl,
            type = request.type,
            sortOrder = request.sortOrder ?: 0,
            createdAt = OffsetDateTime.now(),
        )
    }

    override fun deletePointMedia(
        pointId: Long,
        mediaId: Long,
    ) {
        getPointDetail(pointId)
    }

    companion object {
        private val categories =
            listOf(
                CategoryItem(1L, "Достопримечательности", "attraction"),
                CategoryItem(2L, "Шопинг", "shop"),
                CategoryItem(3L, "Музеи", "museum"),
                CategoryItem(4L, "Парки", "park"),
                CategoryItem(5L, "Еда", "food"),
            )

        private val mock =
            listOf(
                PointShortItem(
                    id = 1L,
                    title = "Красная площадь",
                    shortDescription = "Главная площадь Москвы",
                    categoryId = 1L,
                    categoryName = "Достопримечательности",
                    coordinates = GeoPoint(BigDecimal("55.753544"), BigDecimal("37.621202")),
                    visitTime = 60,
                ),
                PointShortItem(
                    id = 2L,
                    title = "ГУМ",
                    shortDescription = "Главный универсальный магазин",
                    categoryId = 2L,
                    categoryName = "Шопинг",
                    coordinates = GeoPoint(BigDecimal("55.754713"), BigDecimal("37.621500")),
                    visitTime = 90,
                ),
                PointShortItem(
                    id = 3L,
                    title = "Третьяковская галерея",
                    shortDescription = "Крупнейшее собрание русского искусства",
                    categoryId = 3L,
                    categoryName = "Музеи",
                    coordinates = GeoPoint(BigDecimal("55.741426"), BigDecimal("37.620137")),
                    visitTime = 120,
                ),
                PointShortItem(
                    id = 4L,
                    title = "Парк Горького",
                    shortDescription = "Центральный парк культуры и отдыха",
                    categoryId = 4L,
                    categoryName = "Парки",
                    coordinates = GeoPoint(BigDecimal("55.729812"), BigDecimal("37.601348")),
                    visitTime = 150,
                ),
                PointShortItem(
                    id = 5L,
                    title = "Кафе Пушкинъ",
                    shortDescription = "Ресторан русской кухни в стиле XIX века",
                    categoryId = 5L,
                    categoryName = "Еда",
                    coordinates = GeoPoint(BigDecimal("55.764804"), BigDecimal("37.604392")),
                    visitTime = 90,
                ),
            )

        private val mockDetail =
            listOf(
                PointDetailResponse(
                    id = 1L,
                    title = "Красная площадь",
                    description =
                        "Главная площадь Москвы, расположенная в самом центре города между " +
                            "Московским Кремлём и Китай-городом. Включена в список Всемирного наследия ЮНЕСКО.",
                    shortDescription = "Главная площадь Москвы",
                    categoryId = 1L,
                    categoryName = "Достопримечательности",
                    address = "Москва, Красная площадь",
                    coordinates = GeoPoint(BigDecimal("55.753544"), BigDecimal("37.621202")),
                    visitTime = 60,
                    workingHours = "Круглосуточно",
                    media =
                        listOf(
                            PointMediaItem("https://cdn.t-guide.mock/points/1/photo-1.jpg", MediaType.PHOTO, 0),
                            PointMediaItem("https://cdn.t-guide.mock/points/1/photo-2.jpg", MediaType.PHOTO, 1),
                            PointMediaItem("https://cdn.t-guide.mock/points/1/audio-guide.mp3", MediaType.AUDIO, 2),
                        ),
                ),
                PointDetailResponse(
                    id = 2L,
                    title = "ГУМ",
                    description =
                        "Главный универсальный магазин — крупнейший торговый комплекс Москвы, " +
                            "расположенный на Красной площади. Памятник архитектуры псевдорусского стиля.",
                    shortDescription = "Главный универсальный магазин",
                    categoryId = 2L,
                    categoryName = "Шопинг",
                    address = "Москва, Красная площадь, 3",
                    coordinates = GeoPoint(BigDecimal("55.754713"), BigDecimal("37.621500")),
                    visitTime = 90,
                    workingHours = "10:00 - 22:00",
                    media =
                        listOf(
                            PointMediaItem("https://cdn.t-guide.mock/points/2/photo-1.jpg", MediaType.PHOTO, 0),
                            PointMediaItem("https://cdn.t-guide.mock/points/2/photo-2.jpg", MediaType.PHOTO, 1),
                        ),
                ),
                PointDetailResponse(
                    id = 3L,
                    title = "Третьяковская галерея",
                    description =
                        "Художественный музей в Москве, основанный в 1856 году купцом Павлом Третьяковым. " +
                            "Имеет одну из самых крупных в мире коллекций русского изобразительного искусства.",
                    shortDescription = "Крупнейшее собрание русского искусства",
                    categoryId = 3L,
                    categoryName = "Музеи",
                    address = "Москва, Лаврушинский переулок, 10",
                    coordinates = GeoPoint(BigDecimal("55.741426"), BigDecimal("37.620137")),
                    visitTime = 120,
                    workingHours = "10:00 - 18:00, выходной — понедельник",
                    media =
                        listOf(
                            PointMediaItem("https://cdn.t-guide.mock/points/3/photo-1.jpg", MediaType.PHOTO, 0),
                            PointMediaItem("https://cdn.t-guide.mock/points/3/video-tour.mp4", MediaType.VIDEO, 1),
                            PointMediaItem("https://cdn.t-guide.mock/points/3/audio-guide.mp3", MediaType.AUDIO, 2),
                        ),
                ),
                PointDetailResponse(
                    id = 4L,
                    title = "Парк Горького",
                    description =
                        "Центральный парк культуры и отдыха в Москве. Современное городское пространство " +
                            "с набережной, велодорожками, кафе и лекториями.",
                    shortDescription = "Центральный парк культуры и отдыха",
                    categoryId = 4L,
                    categoryName = "Парки",
                    address = "Москва, ул. Крымский Вал, 9",
                    coordinates = GeoPoint(BigDecimal("55.729812"), BigDecimal("37.601348")),
                    visitTime = 150,
                    workingHours = "Круглосуточно",
                    media =
                        listOf(
                            PointMediaItem("https://cdn.t-guide.mock/points/4/photo-1.jpg", MediaType.PHOTO, 0),
                            PointMediaItem("https://cdn.t-guide.mock/points/4/photo-2.jpg", MediaType.PHOTO, 1),
                            PointMediaItem("https://cdn.t-guide.mock/points/4/video-aerial.mp4", MediaType.VIDEO, 2),
                        ),
                ),
                PointDetailResponse(
                    id = 5L,
                    title = "Кафе Пушкинъ",
                    description =
                        "Знаменитый московский ресторан русской кухни в стиле дворянской усадьбы XIX века. " +
                            "Расположен на Тверском бульваре.",
                    shortDescription = "Ресторан русской кухни в стиле XIX века",
                    categoryId = 5L,
                    categoryName = "Еда",
                    address = "Москва, Тверской бульвар, 26А",
                    coordinates = GeoPoint(BigDecimal("55.764804"), BigDecimal("37.604392")),
                    visitTime = 90,
                    workingHours = "Круглосуточно",
                    media =
                        listOf(
                            PointMediaItem("https://cdn.t-guide.mock/points/5/photo-1.jpg", MediaType.PHOTO, 0),
                            PointMediaItem("https://cdn.t-guide.mock/points/5/photo-interior.jpg", MediaType.PHOTO, 1),
                        ),
                ),
            )
    }
}
